### PR TITLE
Add empty data directive warning

### LINKED ISSUE #516: [Warning request] Parameter-less db/dw/dl in ROM0/ROMX sections
As we all know, `db` by itself stands for `ds 1`; the same applies (with the corresponding length adjustments) to `dw` and `dl`. This has been a common source for confusion in ROM sections whenever they do appear, since they'd insert an unexpected byte/word/longword.

It would therefore be very useful to warn when these constructs are used in ROM sections, since they're almost always the result of build tools/macros expecting to generate zero-byte data blocks, not an actual attempt to use `ds 1`, `ds 2` or `ds 4` in ROM. This warning should preferrably be enabled by `-Wall`.

### PR DESCRIPTION
**Pull request by JL2210**
Fixes #516

Also known as parameterless `dX`, this isn't normally intended for ROMs. Enabled by `-Wall`.

### CODE REVIEW

**Comment by aaaaaa123456789** [comment]
I'd reword the warning to "db/dw/dl directive without data", because as written right now, it seems like what is missing is a data directive, not data itself.

Otherwise, LGTM.

**Comment by JL2210** [comment]
@aaaaaa123456789 Done.
